pkg/types/types.go

```diff
@@ -70,7 +70,10 @@ type Address []byte
 
 // UpkeepKey is an identifier of an upkeep at a moment in time, typically an
 // upkeep at a block number
-type UpkeepKey []byte
+type UpkeepKey interface {
+	BlockKeyAndUpkeepID() (BlockKey, UpkeepIdentifier)
```

Comment:
json marshaling should probably also be included here.

Reply:
Do you mean, add a custom marshaling function to the concrete type, or add the json.Unmarshaler interface to this interface?